prime-router/docs/universal-pipeline/route.md

```diff
@@ -1,17 +1,462 @@
+   <td>See code block below
+   </td>
+  </tr>
+</table>
+
+```kotlin
+/**
+ * Default Rules:
+ *   Must have message ID, patient last name, patient first name, DOB, specimen type
+ *   At least one of patient street, patient zip code, patient phone number, patient email
+ *   At least one of order test date, specimen collection date/time, test result date
+ */
+val qualityFilterDefault: ReportStreamFilter = listOf(
+    "%messageId.exists()",
+    "%patient.name.family.exists()",
+    "%patient.name.given.count() > 0",
+    "%patient.birthDate.exists()",
+    "%specimen.type.exists()",
+    "(%patient.address.line.exists() or " +
+        "%patient.address.postalCode.exists() or " +
+        "%patient.telecom.exists())",
+    "(" +
+        "(%specimen.collection.collectedPeriod.exists() or " +
+        "%specimen.collection.collected.exists()" +
+        ") or " +
+        "%serviceRequest.occurrence.exists() or " +
+        "%observation.effective.exists())"
+)
+```
+
+## Routing Filter
+
+Generic filtering that does not concern data quality or condition (e.g. test result is positive)
+
+
+<table>
+  <tr>
+   <td><strong>Topic</strong>
+   </td>
+   <td>Applies to following topics: <em>full-elr, etor-ti, elr-elims</em>
+   </td>
+  </tr>
+  <tr>
+   <td><strong>Operation</strong>
+   </td>
+   <td>expressions are evaluated with AND operation
+   </td>
+  </tr>
+  <tr>
+   <td><strong>Default</strong>
+   </td>
+   <td><strong>Allow All: </strong>the door is open. No filter is in place
+   </td>
+  </tr>
+</table>
+
+
+
+## Processing Mode Code Filter
+
+The processing mode of the data indicates the sender’s intended context for the data. Options for this field are found here [CodeSystem: processingId](https://terminology.hl7.org/5.2.0/CodeSystem-v2-0103.html). The intention is to ensure the sender and receiver operate with the same data content context. Test data should only be accepted by test receivers. Production data should only be accepted by production receivers.
+
+
+<table>
+  <tr>
+   <td><strong>Topic</strong>
+   </td>
+   <td>Applies to following topics: <em>full-elr, etor-ti</em>
+   </td>
+  </tr>
+  <tr>
+   <td><strong>Operation</strong>
+   </td>
+   <td>expressions are evaluated with AND operation
+   </td>
+  </tr>
+  <tr>
+   <td><strong>Default</strong>
+   </td>
+   <td>See code block below
+   </td>
+  </tr>
+</table>
+
+```kotlin
+   /**
+ * Default Rule:
+ *  Must have a processing mode id of 'P'
+ */
+val processingModeFilterDefault: ReportStreamFilter = listOf(
+    "%processingId = 'P'"
+)
+```
+
+## Condition Filter
+
+Filter data based on the test identifiers. A receiver expecting flu results should only accept tests for flu. If the message contains multiple observations, some that pass the condition filter and others that do not, the condition filter will be used to identify the desired observations. Identifiers for the needed observations are added to the Endpoint which is then added to the Provenance resource.
```

Comment:
Is Endpoint the right word here? I am thinking it is something else but am not sure what.

Reply:
Pretty sure Endpoint is right. See FHIRBundleHelpers.addReceivers()